src/main/java/carpet/script/value/EntityValue.java

```diff
@@ -891,21 +893,51 @@ else if (v instanceof ListValue)
             throw new InternalExpressionException("'effect' needs either no arguments (clear) or effect name, duration, and optional amplifier, show particles and show icon");
         });
 
-        // gamemode
-        // spectate
-        // "fire"
-        // "extinguish"
-        // "silent"
-        // "gravity"
-        // "invulnerable"
-        // "dimension"
-        // "item"
-        // "count",
-        // "age",
-        // "effect_"name
-        // "hold"
-        // "hold_offhand"
-        // "jump"
+        put("gamemode_id", (e,v)->{
```

Comment:
rename to 'gamemode' and handle both string as well as numerical values. I would suggest checking variable type and if its a number use the numerical representation, otherwise use getString() and check agains names of gamemodes (case insenstitive)

Reply:
yeah, i tried that at first, but it had a ton of trouble accepting the string I typed in. I had a similar problem when trying to do the cone in my other pr, and finding that I couldnt input 'x','y' or 'z', tho I can't tell why

Reply:
tho i suppose that is a different trouble

Reply:
@gnembon it works only if I use switch statement for some reason